addon/src/model-definition.js

```diff
@@ -4,6 +4,27 @@ import MissingSequenceError from './missing-sequence-error';
 import { isEmptyObject, mergeDeep } from './utils/helper-functions';
 import { assert } from '@ember/debug';
 import { typeOf } from '@ember/utils';
+import { useStringIdsOnly, verifyId } from './own-config';
+import { v4 as uuid } from 'uuid';
+
+/**
+ * A wrapper around generated ids to help with handling them, regardless of string vs int id type.
+ * Keeps track of next id to use for a model definition when creating a new record.
+ */
+class IdGenerator {
+  newId = useStringIdsOnly ? uuid() : 1; // should always be an unused id ready for use
+
+  /**
+   * An unused id is requested. Return the unused id, and cycle it to make a new one. For strings, we can just return a
+   * unique uuid. For ints, we have to increment to the next id, as the only way to ensure it is somewhat-unique for
+   * each record created.
+   */
+  nextId() {
+    const { newId } = this;
+    this.newId = typeof newId === 'string' ? uuid() : newId + 1;
+    return newId;
+  }
```

Comment:
Expecting to drop number ids in next major or so. When that's done, there won't be a need for this class, can just use `uuid()` directly all the time